index.js

```diff
@@ -5,17 +5,18 @@ var stream = require('readable-stream')
 
 inherits(StringStream, stream.Readable)
 
-function StringStream (str) {
-  if (!(this instanceof StringStream)) return new StringStream(str)
+function StringStream (str, encoding) {
+  if (!(this instanceof StringStream)) return new StringStream(str, encoding)
   stream.Readable.call(this)
   this._str = str
+  this._encoding = encoding || 'utf8'
 }
 
 StringStream.prototype._read = function () {
   if (!this.ended) {
     var self = this
     process.nextTick(function () {
-      self.push(new Buffer(self._str))
+      self.push(new Buffer(self._str, self._encoding))
```

Comment:
[`new Buffer` is deprecated according to the docs](https://nodejs.org/api/buffer.html#buffer_new_buffer_string_encoding), so it might be worth accepting the following suggestion.
```suggestion
      self.push(Buffer.from(self._str, self._encoding))
```

Reply:
Yeah, I was going to do that. That change would mean the module is no longer supported in Node versions previous to 5, which we could all probably live with anyway.

Reply:
Yeah that's true, I think it's probably safe to remove support for Node 4 and 5 in newer versions of packages since [they're no longer maintained versions of Node](https://github.com/nodejs/Release) and it's still the user's choice to upgrade this dependency.